wa/commands/runstatus.py

```diff
@@ -0,0 +1,298 @@
+
+    def generate_job_detail(self):
+        detail = ('\n=========================\n'
+                  'Job Detail'
+                  '\n=========================\n\n'
+                  )
+
+        for segment, jobstates in self.segmented.items():
+            if jobstates:
+                for job in jobstates:
+                    detail += ('{} ({}) [{}]'
+                               '{}, {}\n'
+                               ).format(
+                                   job.id,
+                                   job.label,
+                                   job.iteration,
+                                   ' - ' + str(job.retries)if job.retries else '',
+                                   self.formatter.highlight_keyword(str(job.status))
+                                )
+                    job_output = self.job_outputs[(job.id,job.label,job.iteration)]
+                    
+                    for event in job_output.events:
+                        detail += self.formatter.fit_term_width(f'\t{event.summary}\n')
+        return detail
+
+    def generate_run_detail(self):
+        detail = ('\n\n=========================\n'
+                  'Run Events'
+                  '\n=========================\n\n'
+                  ) if self.ro.events else ''
+
+        for event in self.ro.events:
+            detail += f'{event.summary}\n'
+
+        return detail
+
+    def generate_output(self, verbose):
+        output = self.generate_run_header()
+        output += self.generate_job_summary()
+
+        if verbose:
+            output += self.generate_run_detail()
+            output += self.generate_job_detail()
+
+        return output
+
+
+def _seconds_as_smh(seconds):
+    seconds = int(seconds)
+    hours = seconds // 3600
+    minutes = (seconds % 3600) // 60
+    seconds = seconds % 60
+    return seconds, minutes, hours
+
+
+def segment_jobs_by_state(jobstates, max_retries, retry_status):
+    finished_states = [
+            Status.PARTIAL, Status.FAILED,
+            Status.ABORTED, Status.OK, Status.SKIPPED
+    ]
+
+    segmented = {
+        'finished': [], 'other': [], 'running': [],
+        'pending': [], 'uninitialized': []
+    }
+
+    for jobstate in jobstates:
+        if (jobstate.status in retry_status) and jobstate.retries < max_retries:
+            segmented['running'].append(jobstate)
+        elif jobstate.status in finished_states:
+            segmented['finished'].append(jobstate)
+        elif jobstate.status == Status.RUNNING:
+            segmented['running'].append(jobstate)
+        elif jobstate.status == Status.PENDING:
+            segmented['pending'].append(jobstate)
+        elif jobstate.status == Status.NEW:
+            segmented['uninitialized'].append(jobstate)
+        else:
+            segmented['other'].append(jobstate)
+
+    return segmented
+
+
+class _simple_formatter:
+    color_map = {
+        'running': COLOR_MAP[logging.INFO],
+        'ok': colorama.Fore.WHITE,
```

Comment:
Ideally, this should be the default foreground color for the user's terminal, rather than explicitly white. IIRC colorama should have an alias for that? If not, I think jus using `None` and avoiding formatting to  begin with, should be way to go. Do not assume that people use white as their foreground color (e.g. this will work terribly on terminals with white bacground and dark foreground).